core/src/main/scala/cats/Applicative.scala

```diff
@@ -200,7 +200,7 @@ object Applicative {
    * scala> import cats.Applicative.catsApplicativeForArrow
    * scala> val toLong: Int => Long = _.toLong
    * scala> val double: Int => Int = 2*_
-   * scala> val f: Int => (Long, Int) = catsApplicativeForArrow.product(toLong, double)
+   * scala> val f: Int => (Long, Int) = catsApplicativeForArrow[Function1, Int].product(toLong, double)
```

Comment:
since `PartialFunction` is subtype of `Function1` there was implicit ambiguity here